lib/foreman_maintain/utils/backup.rb

```diff
@@ -241,6 +241,24 @@ def validate_hostname?
         end
       end
 
+      def validate_interfaces?
+        # I wanted to do `Socket.getifaddrs.map(&:name).uniq`,
+        # but this has to work with Ruby 2.0, and Socket.getifaddrs is 2.1+
+        system_interfaces = Dir.children('/sys/class/net')
```

Comment:
Maybe we can do something like below, return invalid interfaces and actual check will confirm if hash is empty?
~~~
        system_interfaces = Dir.children('/sys/class/net')
        invalid_interfaces = {}
        proxy_config = metadata.fetch('proxy_config', {})
        dns_dhcp_config = ( ['dns', 'dhcp'] & proxy_config.keys )
        if dns_dhcp_config.any?
          dns_dhcp_config.each do |interface|
            unless system_interfaces.include?(proxy_config.fetch(interface, 'lo'))
              invalid_interfaces[interface] = proxy_config.fetch(interface, 'lo')
            end
          end
        end
        invalid_interfaces
~~~